qcodes/dataset/plotting.py

```diff
@@ -68,6 +69,9 @@ def plot_by_id(run_id: int,
         colorbar axes may be None if no colorbar is created (e.g. for
         1D plots)
     """
+    # defaults
+    smart_colorscale = smart_colorscale or config.gui.smart_colorscale
```

Comment:
Is this logic correct?
Defalt value:
None or True => True
None or False => False
True or False => True
True or True => True 
False or False => True
Seems correct 
but 
False or True  => True
seems wrong to me?

Reply:
That is indeed a very good point.